src/library/scala/util/Random.scala

```diff
@@ -38,7 +38,15 @@ class Random(val self: java.util.Random) extends AnyRef with Serializable {
    *  array.
    */
   def nextBytes(bytes: Array[Byte]): Unit = { self.nextBytes(bytes) }
-
+  
+  /** Generates `n` random bytes and returns them in a new array. */
+  def nextBytes(n: Int): Array[Byte] = {
+    require(n > 0, "n must be positive")
```

Comment:
You shouldn't require anything.  This should be like `fill` which just returns an empty array on 0 or negative numbers.

Reply:
`Array.ofDim` blows up for negative numbers:
```scala
scala> Array.ofDim[Int](-1)
java.lang.NegativeArraySizeException
  at scala.reflect.ManifestFactory$IntManifest.newArray(Manifest.scala:123)
  at scala.reflect.ManifestFactory$IntManifest.newArray(Manifest.scala:121)
  at scala.Array$.ofDim(Array.scala:217)
  ... 28 elided
```
Should I bring back the require with >= 0? Or should `ofDim` be changed?

Reply:
```
if (n < 1) new Array[Byte](0)
else {
  val bytes new Array[Byte](n)
   ...
```
is reasonable.  I'm not sure `ofDim` adds anything over the direct call to the `Array[Byte]` constructor.